scene/debugger/scene_debugger.cpp

```diff
@@ -1965,9 +2345,184 @@ void RuntimeNodeSelect::_find_3d_items_at_pos(const Point2 &p_pos, Vector<Select
+	// Get the camera normal.
+	if (cam_override) {
+		near_plane = Plane(root->get_camera_3d_override_transform().basis.get_column(2), cam_pos);
+	} else {
+		near_plane = Plane(camera->get_global_transform().basis.get_column(2), cam_pos);
+	}
+	near_plane.d -= znear;
+	frustum.push_back(near_plane);
+
+	Plane far_plane = -near_plane;
+	far_plane.d += zfar;
+	frustum.push_back(far_plane);
+
+	Vector<Vector3> points = Geometry3D::compute_convex_mesh_points(&frustum[0], frustum.size());
+	Ref<ConvexPolygonShape3D> shape;
+	shape.instantiate();
+	shape->set_points(points);
+
+	// Start with physical objects.
+	PhysicsDirectSpaceState3D *ss = root->get_world_3d()->get_direct_space_state();
+	PhysicsDirectSpaceState3D::ShapeResult result;
+	PhysicsDirectSpaceState3D::ShapeParameters shape_params;
+	shape_params.shape_rid = shape->get_rid();
+	shape_params.collide_with_areas = true;
+	if (ss->intersect_shape(shape_params, &result, 32)) {
+		SelectResult res;
+		res.item = Object::cast_to<Node>(result.collider);
+		res.order = -dist_pos.distance_to(Object::cast_to<Node3D>(res.item)->get_global_transform().origin);
+
+		// Fetch collision shapes.
+		CollisionObject3D *collision = Object::cast_to<CollisionObject3D>(result.collider);
+		if (collision) {
+			List<uint32_t> owners;
+			collision->get_shape_owners(&owners);
+			for (const uint32_t &I : owners) {
+				SelectResult res_shape;
+				res_shape.item = Object::cast_to<Node>(collision->shape_owner_get_owner(I));
+				res_shape.order = res.order;
+				r_items.push_back(res_shape);
+			}
+		}
+
+		r_items.push_back(res);
+	}
+
+	// Then go for the meshes.
+	Vector<ObjectID> items = RS::get_singleton()->instances_cull_convex(frustum, root->get_world_3d()->get_scenario());
+	for (int i = 0; i < items.size(); i++) {
+		Object *obj = ObjectDB::get_instance(items[i]);
+		GeometryInstance3D *geo_instance = nullptr;
+		Ref<TriangleMesh> mesh_collision;
+
+		MeshInstance3D *mesh_instance = Object::cast_to<MeshInstance3D>(obj);
+		if (mesh_instance) {
+			if (mesh_instance->get_mesh().is_valid()) {
+				geo_instance = mesh_instance;
+				mesh_collision = mesh_instance->get_mesh()->generate_triangle_mesh();
+			}
+		} else {
+			Label3D *label = Object::cast_to<Label3D>(obj);
+			if (label) {
+				geo_instance = label;
+				mesh_collision = label->generate_triangle_mesh();
+			} else {
+				Sprite3D *sprite = Object::cast_to<Sprite3D>(obj);
+				if (sprite) {
+					geo_instance = sprite;
+					mesh_collision = sprite->generate_triangle_mesh();
+				}
+			}
+		}
+
+		if (mesh_collision.is_valid()) {
+			Transform3D gt = geo_instance->get_global_transform();
+			if (mesh_collision->inside_convex_shape(frustum.ptr(), frustum.size(), points.ptr(), points.size(), gt.get_basis().get_scale())) {
+				SelectResult res;
+				res.item = Object::cast_to<Node>(obj);
+				res.order = -dist_pos.distance_to(gt.origin);
+				r_items.push_back(res);
+
+				continue;
+			}
+		}
+
+		items.remove_at(i);
+		i--;
+	}
```

Comment:
@Arnklit asked me to take a look and figure out why the drag select wasn't working in this PR. The problem comes from this loop. The `items` array is correct and captures all the objects, but this loop incorrectly culls objects that are within the boundary of the selection. 

Replacing this loop with the following fixes it completely (and is way faster):
```
for (int i = 0; i < items.size(); i++) {
	Object *obj = ObjectDB::get_instance(items[i]);

	Node3D *instance = Object::cast_to<Node3D>(obj);
	if (instance) {
		SelectResult res;
		res.item = Object::cast_to<Node>(obj);
		res.order = -dist_pos.distance_to(instance->get_global_position());
		r_items.push_back(res);
	}
}
```